driver-core/src/main/com/mongodb/connection/CompositeByteBuf.java

```diff
@@ -41,7 +41,7 @@
 
         int offset = 0;
         for (ByteBuf cur : buffers) {
-            Component component = new Component(cur.duplicate().order(ByteOrder.LITTLE_ENDIAN), offset);
+            Component component = new Component(cur.asReadOnly().order(ByteOrder.LITTLE_ENDIAN), offset);
```

Comment:
In netty it does as it returns `cur` because there is no readOnly concept, in `ByteBufNIO` it returns a new instance with a reference of 1.

The resulting `ByteBuf` is later released by `ByteBufferBsonOutput`. It's not a great situation.